fri/src/prover/mod.rs

```diff
@@ -89,36 +91,41 @@ mod tests;
 ///
 /// Calling [build_layers()](FriProver::build_layers()) when the internal state is dirty, or
 /// calling [build_proof()](FriProver::build_proof()) on a clean state will result in a panic.
-pub struct FriProver<B, E, C, H>
+pub struct FriProver<B, E, C, H, V>
 where
     B: StarkField,
     E: FieldElement<BaseField = B>,
-    C: ProverChannel<E, Hasher = H>,
+    C: ProverChannel<E, H, Hasher = H>,
     H: ElementHasher<BaseField = B>,
+    V: VectorCommitment<H>,
 {
     options: FriOptions,
-    layers: Vec<FriLayer<B, E, H>>,
+    layers: Vec<FriLayer<B, E, H, V>>,
     remainder_poly: FriRemainder<E>,
     _channel: PhantomData<C>,
 }
 
-struct FriLayer<B: StarkField, E: FieldElement<BaseField = B>, H: Hasher> {
-    tree: MerkleTree<H>,
+struct FriLayer<B: StarkField, E: FieldElement<BaseField = B>, H: Hasher, V: VectorCommitment<H>> {
+    commitment: V,
     evaluations: Vec<E>,
     _base_field: PhantomData<B>,
+    _h: PhantomData<H>,
 }
 
 struct FriRemainder<E: FieldElement>(Vec<E>);
 
 // PROVER IMPLEMENTATION
 // ================================================================================================
 
-impl<B, E, C, H> FriProver<B, E, C, H>
+impl<B, E, C, H, V> FriProver<B, E, C, H, V>
 where
     B: StarkField,
     E: FieldElement<BaseField = B>,
-    C: ProverChannel<E, Hasher = H>,
+    C: ProverChannel<E, H, Hasher = H, VectorCommitment = V>,
     H: ElementHasher<BaseField = B>,
+    V: VectorCommitment<H>,
+    <V as VectorCommitment<H>>::Item: From<<H as Hasher>::Digest>,
+    <V as VectorCommitment<H>>::Commitment: From<<H as Hasher>::Digest>,
```

Comment:
This basically requires that both `VectorCommitment::Item` and `VectorCommitment::Commitment` are digests, right?

Reply:
Same reply above applies